### ISSUE TITLE
Trivial CATCH / re-THROW

### ISSUE THREAD

**Issue by mikefechner** [Riverside-Software/sonar-openedge#225]
The default templates used by the Visual Designer incorporate the following in the constructor:

```
CATCH e AS Progress.Lang.Error:
    UNDO, THROW e .
END CATCH .
```

When the class is using ```ROUTINE-LEVEL UNDO, THROW``` or ```BLOCK-LEVEL UNDO, THROW``` this is 

a) not needed as the uncatched error would be thrown further anyway
b) the StackTrace of the error is changed from the original issue location to the re-THROW - which makes debugging harder. 

Therefore in general a CATCH with an unconditional re-THROW should be reported as a rule violation.

**Comment by gquerret** [comment]
Just to confirm the behavior, would a `CATCH xxx: UNDO, THROW xxx. END CATCH.` would be reported only when `ROUTINE-LEVEL UNDO, THROW` or `BLOCK-LEVEL` is present ? Or always ?

**Comment by mikefechner** [comment]
> would be reported only when ROUTINE-LEVEL UNDO, THROW or BLOCK-LEVEL is present ? Or always ?

Correct. In the case below the "trivial CATCH - rethrow" is useful. 

But if we add BLOCK-LEVEL or ROUTINE-LEVEL to the mix it isn't. 

```
RUN test .

CATCH err AS Progress.Lang.Error :
    MESSAGE "catch:" err:GetMessage(1)
        VIEW-AS ALERT-BOX.
END CATCH.

PROCEDURE test:

    DEFINE VARIABLE h AS HANDLE NO-UNDO.

    h:QUERY-OPEN () .

    CATCH err AS Progress.Lang.Error :
        UNDO, THROW err .
    END CATCH.

END.
```

**Comment by gquerret** [comment]
New rule created in develop branch, will be included in next release
